File: BichromeBoard/solve/cpp/BichromeBoardSolver.cpp
#include <fstream>
#include <topcoder>
#include "BichromeBoard.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
    try {
        std::ifstream ifs(argv[1]);
        vector<string> board; tc::read(ifs, board);
        ifs.close();

        std::ofstream ofs(argv[2]);
        BichromeBoard solver;
        tc::show(ofs, solver.ableToDraw(board));
        ofs.close();
    } catch (std::exception &e) {
        std::cerr << e.what() << std::endl;
    }
    return 0;
}

File: BichromeBoard/solve/python/BichromeBoardSolver.py
#! /usr/bin/env python
import os, sys
import BichromeBoard

def init():
    default_path = os.path.join(os.getenv("HOME"), ".gettc")
    gettc_home = os.path.abspath(os.getenv("GETTC_HOME", default_path))
    include_dir = os.path.join(gettc_home, "include/python")
    sys.path.append(include_dir)

def main():
    import topcoder as tc
    with open(sys.argv[1], "r") as fi:
        input = fi.read()
        reader = tc.Reader(input)
    
        board = reader.next("String[]")

    result = BichromeBoard.ableToDraw(board)
    with open(sys.argv[2], "w") as fo:
        fo.write(tc.write(result, "String"))

if __name__ == "__main__":
    init()
    main()
